File: Daily Challanges/2026-06-June/2026-06-05.py
def is_valid_schema(obj):
    roles = {"user", "creator", "moderator", "staff", "admin"}
    required = ["username", "posts", "verified", "role", "badges"]
    if not all(key in obj for key in required):
        return False
    if not isinstance(obj["username"], str):
        return False
    if not isinstance(obj["posts"], (int, float)) or isinstance(obj["posts"], bool):
        return False
    if not isinstance(obj["verified"], bool):
        return False
    if obj["role"] not in roles:
        return False
    if "supporter" in obj and not isinstance(obj["supporter"], bool):
        return False
    if not isinstance(obj["badges"], list):
        return False
    if not all(isinstance(badge, str) for badge in obj["badges"]):
        return False
    return True


print(is_valid_schema({"username": "gill", "posts": 12, "verified": False, "role": "creator", "supporter": False, "badges": ["early-adopter", "popular"]}))
print(is_valid_schema({"username": "tonya", "posts": 299, "verified": True, "role": "moderator", "supporter": True, "badges": ["streak-master", "veteran"], "followers": 1233}))
print(is_valid_schema({"username": "zara", "posts": 0, "verified": False, "role": "user", "supporter": False, "badges": []}))
print(is_valid_schema({"username": "nicole", "posts": 65, "verified": True, "role": "admin", "supporter": False, "badges": ["first-post", 18]}))
print(is_valid_schema({"username": "tim", "posts": 25, "verified": True, "role": "staff", "supporter": False}))
print(is_valid_schema({"username": "charlie", "posts": 0, "verified": False, "role": "user", "supporter": "no", "badges": ["first-post", "anniversary"]}))
print(is_valid_schema({"username": "wanda", "posts": 15, "verified": True, "role": "friend", "supporter": True, "badges": ["popular"]}))
print(is_valid_schema({"username": "guy", "posts": 5, "verified": "false", "role": "staff", "supporter": True, "badges": ["helper"]}))
print(is_valid_schema({"username": "carrie", "verified": True, "role": "moderator", "supporter": True, "badges": ["helper", "sharer"]}))
print(is_valid_schema({"username": True, "posts": 75, "verified": True, "role": "creator", "supporter": True, "badges": ["veteran"]}))

File: Daily Challanges/2026-06-June/2026-06-05.js
function isValidSchema(obj) {
  const roles = new Set(["user", "creator", "moderator", "staff", "admin"]);
  const required = ["username", "posts", "verified", "role", "badges"];

  if (!required.every((key) => key in obj)) {
    return false;
  }

  if (typeof obj.username !== "string") {
    return false;
  }

  if (typeof obj.posts !== "number") {
    return false;
  }

  if (typeof obj.verified !== "boolean") {
    return false;
  }

  if (!roles.has(obj.role)) {
    return false;
  }

  if ("supporter" in obj && typeof obj.supporter !== "boolean") {
    return false;
  }

  if (!Array.isArray(obj.badges)) {
    return false;
  }

  if (!obj.badges.every((badge) => typeof badge === "string")) {
    return false;
  }

  return true;
}

console.log(
  isValidSchema({
    username: "gill",
    posts: 12,
    verified: false,
    role: "creator",
    supporter: false,
    badges: ["early-adopter", "popular"],
  }),
);
console.log(
  isValidSchema({
    username: "tonya",
    posts: 299,
    verified: true,
    role: "moderator",
    supporter: true,
    badges: ["streak-master", "veteran"],
    followers: 1233,
  }),
);
console.log(
  isValidSchema({
    username: "zara",
    posts: 0,
    verified: false,
    role: "user",
    supporter: false,
    badges: [],
  }),
);
console.log(
  isValidSchema({
    username: "nicole",
    posts: 65,
    verified: true,
    role: "admin",
    supporter: false,
    badges: ["first-post", 18],
  }),
);
console.log(
  isValidSchema({
    username: "tim",
    posts: 25,
    verified: true,
    role: "staff",
    supporter: false,
  }),
);
console.log(
  isValidSchema({
    username: "charlie",
    posts: 0,
    verified: false,
    role: "user",
    supporter: "no",
    badges: ["first-post", "anniversary"],
  }),
);
console.log(
  isValidSchema({
    username: "wanda",
    posts: 15,
    verified: true,
    role: "friend",
    supporter: true,
    badges: ["popular"],
  }),
);
console.log(
  isValidSchema({
    username: "guy",
    posts: 5,
    verified: "false",
    role: "staff",
    supporter: true,
    badges: ["helper"],
  }),
);
console.log(
  isValidSchema({
    username: "carrie",
    verified: true,
    role: "moderator",
    supporter: true,
    badges: ["helper", "sharer"],
  }),
);
console.log(
  isValidSchema({
    username: true,
    posts: 75,
    verified: true,
    role: "creator",
    supporter: true,
    badges: ["veteran"],
  }),
);
